src/contracts/Staking.sol

```diff
@@ -311,32 +354,30 @@ contract Staking is Ownable {
     }
 
     /**
-        @notice claims STAKING_TOKEN after cooldown period
-        @param _recipient address
+        @notice claims staking tokens after cooldown period
+        @dev if user has a cooldown claim that's past expiry then withdraw staking tokens from tokemak
+        @dev and send them to user
+        @param _recipient address - users unstaking address
      */
     function claimWithdraw(address _recipient) public {
         Claim memory info = coolDownInfo[_recipient];
 
+        ITokeManager tokeManager = ITokeManager(TOKE_MANAGER);
         ITokePool tokePoolContract = ITokePool(TOKE_POOL);
-        WithdrawalInfo memory withdrawalInfo = tokePoolContract
+        RequestedWithdrawalInfo memory requestedWithdrawals = tokePoolContract
             .requestedWithdrawals(address(this));
         uint256 totalAmountIncludingRewards = IRewardToken(REWARD_TOKEN)
             .balanceForGons(info.gons);
+        uint256 currentCycleIndex = tokeManager.getCurrentCycleIndex();
         if (
-            (_isClaimAvailable(info)) &&
-            withdrawalInfo.amount >= totalAmountIncludingRewards
+            _isClaimAvailable(info) &&
+            requestedWithdrawals.minCycle <= currentCycleIndex &&
+            requestedWithdrawals.amount >= totalAmountIncludingRewards
         ) {
             _withdrawFromTokemak(totalAmountIncludingRewards);
 
-            // revert if not enough funds to cover transfer
-            uint256 stakingBalance = IERC20(STAKING_TOKEN).balanceOf(
-                address(this)
-            );
-
-            // must have enough funds to withdraw
-            require(stakingBalance >= info.amount, "Not enough funds");
```

Comment:
I removed this because it's already checked by line 375.  I renamed the variables because it wasn't self explanatory